### PR TITLE
Expand subnet value predicates

### PR DESCRIPTION
**Pull request by mavam**
###  :notebook_with_decorative_cover: Description

This PR expands value predicates of type `subnet` as follows: given a subnet `S`, the parser expands it to `:subnet == S` and the normalization then further to `:subnet == S || :addr in S`.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File by file.

### CODE REVIEW

**Review by dominiklohmann — CHANGES_REQUESTED** [review]
There's a problem with the current implementation.

Instead of expanding just value predicates, it expands any expression `:subnet == <subnet>`, making it impossible to query just for subnets by equality. 

In a call with @mavam we discussed two possible options:
- Move the expansion into the parser.
- Make value predicates valid predicates, i.e., don't expand them during parsing at all.

The second one would be a larger refactoring, so we'll likely opt for the first. The documentation should be rewritten accordingly to make clear this only affects value predicates.

File: libvast/src/expression_visitors.cpp
Comment (dominiklohmann) [inline libvast/src/expression_visitors.cpp:60]: This function urgently needs comments above every step. Make it easy for the future maintainer to follow this function. `make_addr_pred` as a function name alone is not sufficient here.

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (dominiklohmann) [inline libvast/src/concept/parseable/vast/expression.cpp:23]: The expander doesn't really belong in `expression_visitor.hpp`. Can you move it to this file, remove the include, and rename it to something more specific like `value_predicate_expander`?

File: libvast/vast/expression_visitors.hpp
Comment (dominiklohmann) [inline libvast/vast/expression_visitors.hpp:32]: Comment does not reflect what this actually does.

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (dominiklohmann) [inline libvast/src/concept/parseable/vast/expression.cpp:190]: Should we just call this inside of `to_value_predicate`? Why add another lambda on top?

**Review by dominiklohmann — COMMENTED** [review]
Just some nits. I ran this locally and it works fine on my end.

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (dominiklohmann) [reply libvast/src/concept/parseable/vast/expression.cpp:190]: Or, the other way around: Should we call `to_value_predicate` inside the expander? Then its name makes sense.

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (dominiklohmann) [inline libvast/src/concept/parseable/vast/expression.cpp:65]: Why aren't there empty lines between the member functions?

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (dominiklohmann) [inline libvast/src/concept/parseable/vast/expression.cpp:60]: Is doxygenize a word? If so, please doxygenize the comment by adding another `/`.

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (dominiklohmann) [inline libvast/src/concept/parseable/vast/expression.cpp:111]: Please `s/^/\/` here as well.

**Review by dominiklohmann — APPROVED** [review]
Approved modulo some minor fixups.

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (mavam) [reply libvast/src/concept/parseable/vast/expression.cpp:66]: Fixed.

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (mavam) [reply libvast/src/concept/parseable/vast/expression.cpp:60]: Disagreeing here; this is an implementation detail local to the function. It will never land in documentation.

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (mavam) [reply libvast/src/concept/parseable/vast/expression.cpp:114]: s.a.

File: libvast/src/concept/parseable/vast/expression.cpp
Comment (dominiklohmann) [reply libvast/src/concept/parseable/vast/expression.cpp:60]: Tooling disagrees with you. I get a different hint from clangd on autocomplete with three slashes.

Comment (mavam) [reply libvast/src/concept/parseable/vast/expression.cpp:60]: Thank god I'm a human and can make decisions with more context. 😉

Comment (mavam) [reply libvast/src/concept/parseable/vast/expression.cpp:60]: Joking aside, this visitor is in an anonymous namespace. It should not be publicly documented.

Comment (mavam) [reply libvast/src/concept/parseable/vast/expression.cpp]: Per our call, turns out the de-facto standard for language servers is to use triple slashes...who would have known.

**Pull request merged**
